mirdata/datasets/openmic2018.py

```diff
@@ -0,0 +1,370 @@
+
+    @property
+    def title(self) -> Optional[str]:
+        """The title of the track.
+
+        Returns:
+            * title
+        """
+        return self._track_metadata.get("track_title")
+
+    @property
+    def url(self) -> Optional[str]:
+        """The URL on Free Music Archive for this track
+
+        Returns:
+            * str - URL
+        """
+        return self._track_metadata.get("track_url")
+
+    @property
+    def start_time(self) -> float:
+        """The starting time (in seconds) of the selected excerpt within the track.
+
+        Returns:
+            * float - starting time
+        """
+        return self._track_metadata.get("start_time")
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """The track's audio
+
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    @property
+    def vggish(self) -> Tuple[np.ndarray, np.ndarray]:
+        """The track's pre-computed VGGish features
+
+        Returns:
+            * np.ndarray - time indices (seconds) for each frame
+            * np.ndarray - VGGish features; shape=(n_frames, 128)
+        """
+
+        with open(self.vggish_path, "r") as fd:
+            data = json.load(fd)
+            times = np.asarray(data["time_points"])
+            vgg = np.asarray(data["features"])
+            return times, vgg
+
+    # -- will be fed as beat_data=[(self.beats, None)], see jams_utils), and returns a jams
+    # -- object with the annotations.
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            metadata=self._track_metadata,
+            # tag_data=None,  # FIXME
+        )
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO) -> Tuple[np.ndarray, float]:
+    """Load an OpenMIC2018 audio file.
+
+    Audio will be resampled to 44100 Hz and downmixed to mono.
+
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to an audio file
+
+    Returns:
+        * np.ndarray - the audio signal
+        * float - The sample rate of the audio file
+
+    """
+    # -- load as 44100 mono
+    return librosa.load(fhandle, sr=44100, mono=True)
+
+
+# -- use this decorator so the docs are complete
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """The OpenMIC-2018 dataset"""
+
+    def __init__(self, data_home=None, version="default"):
+        super().__init__(
+            data_home,
+            version,
+            name="openmic2018",
+            track_class=Track,
+            bibtex=BIBTEX,
+            indexes=INDEXES,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.cached_property
+    def _metadata(self):
+        metadata_path = Path(self.data_home) / "openmic-2018-metadata.csv"
```

Comment:
could this fail on Windows? maybe replace with os.path.join?

Reply:
similar issues below

Reply:
No, it won't fail on windows.  Pathlib implements `__div__` as syntactic sugar for something like os.path.join, and is generally the more modern way to do it.